### PR TITLE
ci: conditional execution of tests on file changes

### PR DESCRIPTION
**Pull request by jthegedus**
# Summary

This is a test of conditional execution of GitHub Workflows. Namely for our Bats tests and compilation of our documentation site during PRs, but only when files pertaining to their implementation changes.

## Other Information

Also introduce a test case for our documentation site. Currently, since the site build only happens during the deployment script, and only on the master branch commits, it is difficult to identify if there are any issues without watching the master branch updates. It would be good to get these warnings early at the PR stage as it is gated against workflow failures.

This was the initial motivation for conditional executions.

### CODE REVIEW

**Comment by jthegedus** [comment]
It worked! The `paths-filter` action diffed the PR to the base correctly, identified filter matches correctly, and executed the workflows as expected.

<details>
<summary>paths-filter action output</summary>

```
Run dorny/paths-filter@v[2](https://github.com/asdf-vm/asdf/actions/runs/3980533850/jobs/6823679138#step:3:2)
  with:
    filters: documentation:
    - '.github/workflows/**'
    - 'docs/**'
    - '.tool-versions'
  cli:
    - '.github/workflows/**'
    - 'bin/**'
    - 'lib/**'
    - 'scripts/**'
    - 'test/**'
    - '.tool-versions'
    - 'asdf.elv'
    - 'asdf.fish'
    - 'asdf.nu'
    - 'asdf.sh'
  
    token: ***
    list-files: none
    initial-fetch-depth: 100
  env:
    ELVISH_VERSION: v0.18.0
    NUSHELL_VERSION: 0.7[3](https://github.com/asdf-vm/asdf/actions/runs/3980533850/jobs/6823679138#step:3:3).0
Fetching list of changed files for PR#1[4](https://github.com/asdf-vm/asdf/actions/runs/3980533850/jobs/6823679138#step:3:4)4[8](https://github.com/asdf-vm/asdf/actions/runs/3980533850/jobs/6823679138#step:3:8) from Github API
  Invoking listFiles(pull_number: 1448, per_page: [10](https://github.com/asdf-vm/asdf/actions/runs/3980533850/jobs/6823679138#step:3:10)0)
  Received 3 items
  [renamed] .github/workflows/documentation.yml
  [modified] .github/workflows/semantic-pr.yml
  [modified] .github/workflows/tests.yml
Detected 4 changed files
Results:
Filter documentation = true
  Matching files:
  .github/workflows/documentation.yml [added]
  .github/workflows/docs.yml [deleted]
  .github/workflows/semantic-pr.yml [modified]
  .github/workflows/tests.yml [modified]
Filter cli = true
  Matching files:
  .github/workflows/documentation.yml [added]
  .github/workflows/docs.yml [deleted]
  .github/workflows/semantic-pr.yml [modified]
  .github/workflows/tests.yml [modified]
Changes output set to ["documentation","cli"]
```

</details>

One of the issues I have had with conditional filtering in the past is that depending on the implementation, filters could only see the files from the latest pushed commit, not the files from the latest commit diffed against the PR base branch. So in the final commit to a PR, one could simply push a yaml file change to get a green check for yaml linting, and the codebase tests wouldn't run 🙄 

Nonetheless, this seems to work well here.

**Comment by jthegedus** [comment]
Even after a master branch merge as the latest commit executed the correct workflows 😄

**Comment by jthegedus** [comment]
Another hope would be that I don't have to wait for macOS test suite to run when making documentation changes.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -73,3 +132,48 @@ jobs:
...
+        with:
+          # fetch all commits to get git log info for Vuepress
+          fetch-depth: 0
+
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            docs:
+              - 'docs/**'
```
Comment (Stratus3D) [inline .github/workflows/tests.yml:154]: Where is the output from this step used?

Comment (jthegedus) [reply .github/workflows/tests.yml:154]: Ah, yes, that was from prior testing of the filter action. I will remove.

**Pull request merged**